posts/2024-07-30-24.0.0.8-beta.adoc

```diff
@@ -0,0 +1,816 @@
+
+This is an early beta release to support the ratification of MicroProfile 7.0. As such, some function which was available in previous versions of the feature are not yet implemented:
+
+- Documenting more than one web module as link:https://openliberty.io/docs/latest/documentation-openapi.html#multi-module[can be configured in previous versions]
+- Validation of the produced OpenAPI document
+
+=== Further resources:
+
+For more information, see the following resources:
+
+- link:https://download.eclipse.org/microprofile/microprofile-open-api-4.0-RC4/microprofile-openapi-spec-4.0-RC4.html[MP OpenAPI 4.0 RC4 specification document] and link:https://download.eclipse.org/microprofile/microprofile-open-api-4.0-RC4/apidocs/[API Javadoc].
+- link:https://openliberty.io/docs/latest/documentation-openapi.html[API documentation with OpenAPI]
+- Guide: link:https://openliberty.io/guides/microprofile-openapi.html[Documenting RESTful APIs]
+
+// DO NOT MODIFY THIS LINE. </GHA-BLOG-TOPIC>
+
+// // // // DO NOT MODIFY THIS COMMENT BLOCK <GHA-BLOG-TOPIC> // // // //
+// Blog issue: https://github.com/OpenLiberty/open-liberty/issues/28869
+// Contact/Reviewer: KyleAure,njr-11
+// // // // // // // //
+
+[#data]
+== Bring your own Jakarta Data provider with Jakarta Data API 1.0
+
+Jakarta Data 1.0 is a new Jakarta EE specification that standardizes a repository-based programming model for data access across relational and non-relational databases.
+
+The Jakarta Data API feature (`dataContainer-1.0`) enables you to use third-party Jakarta Data providers in Open Liberty without including the built-in provider for EclipseLink. This configuration is useful if you want to use a different Jakarta Persistence-based Jakarta Data provider, such as Hibernate, without colliding on the Jakarta Persistence Entity annotation. It's also useful if you want to use Jakarta NoSQL exclusively and do not want the overhead of the built-in provider for relational databases.
+
+To use a third-party Jakarta Data provider, enable the `dataContainer-1.0` feature in your `server.xml` file. You can then include a third-party Jakarta Data provider in your server configuration by using the `library` element.
+
+The following `server.xml` file example shows the configuration to use the Eclipse jNoSQL Jakarta Data provider and a MongoDB database:
+
+[source,xml]
+----
+  <featureManager>
+    <feature>dataContainer-1.0</feature>
+
+    <!-- Features needed for jNoSQL config / processing -->
+    <feature>mpConfig-3.1</feature>
+    <feature>jsonb-3.0</feature>
+  </featureManager>
+
+  <library id="jnosql">
+  	<!-- Jakarta NoSQL API -->
+  	<fileset dir="${shared.resource.dir}/nosql" includes="jakarta.nosql-api.jar" />
+  	<!-- Eclipse jNoSQL Implementation of Jakarta NoSQL and Jakarta Data -->
+    <fileset dir="${shared.resource.dir}/jnosql"
+    	includes="jnosql-communication-core.jar jnosql-mapping-core.jar jnosql-mongodb.jar *.jar" />

```

Comment:
@KyleAure wrote this example and I believe also tried them out. Kyle - could you confirm if other jars were needed in this case, and in the case of Jared's subsequent comment.

Reply:
@njr-11  @jhanders34 Yes, there are transitive dependencies needed by the jnosql drivers.  I figured this would be a good way to represent "Here are the drivers you need, but also include all their dependencies as well."